### ISSUE TITLE
[KEYBOARD]: VAOS - Upcoming and past appointment tabs MUST handle keyboard navigation and data properly

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#8157]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The upcoming and past appointments links are marked up as ARIA tabs, which is a strong interface selection for this type of presentation. They need to handle keyboard focus and arrow key events as laid out in the WCAG WAI-ARIA authoring practices spec. I've found a good article on Inclusive Components that outlines the spec minus a couple of details, which I'll add in the success criteria.

[Inclusive Components: True Tabbed Interfaces](https://inclusive-components.design/tabbed-interfaces/#truetabbedinterfaces)
[Stefan Judis: What's roving tabindex?](https://www.stefanjudis.com/today-i-learned/roving-tabindex/)
[WCAG 2.0 Success Criteria 2.1.1: Keyboard](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [x] Confirm the Past appointments tab shows when users copy/paste `/past` onto the existing URL. Talked with @jbalboni this is how the application behaves now, and is a good behavior for large data fetches.
- [x] The left and right arrow keys should toggle the selected tab. Tabs are treated like radio buttons in the spec, using a roving tab index.
- [x] Pressing the down arrow should set focus on the selected tab panel
- [x] Tab content should be in a div with `role="tabpanel"` for screen readers, per the Inclusive Components guidance
- ~~The past appointments should be loaded async in the background as soon as possible, so when users click or arrow to the second tab, the data is ready instantly.~~

**Comment by jbalboni** [comment]
> The tabbed interface should not change the URL to /past when users select Past Appointments. This breaks the React app and should be prevented defensively.

It's frustrating to have your app called broken when it's not broken and is following a pattern laid out in vets-website code (i.e. claim status). I understand that accessibility guidance can change (claim status and appeals have passed accessibility audits in their current state, as far as I know), and we'll update VAOS to match whatever requirements you have, but it's not a good experience as a VFS team for the only example code/pattern we see for tabs be something that is apparently bad practice.

> The past appointments should be loaded async in the background as soon as possible, so when users click or arrow to the second tab, the data is ready instantly.

This could waste data for users on slow/data capped connections who don't end up looking at the past appointments data. I think we'd want to make the decision to do this based on usage numbers post-launch.

**Comment by 1Copenut** [comment]
Thanks again @jbalboni for the quick conversation around expected behavior. I updated the first acceptance criteria to reflect application behavior and showing the Past appointments tab if users copy/paste the string `/past` to the URL and press ENTER again. This will help out for those on metered data plans too, by only fetching data as it's needed.

I also struck the AC about loading data async in the background. LMK if any of the criteria need tweaked and I'll update after short.
